contenttype/utils.go

```diff
@@ -0,0 +1,39 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package contenttype
+
+import "strings"
+
+const (
+	// CloudEventContentType is the content type for cloud event.
+	CloudEventContentType = "application/cloudevents+json"
+	// JSONContentType is the content type for JSON.
+	JSONContentType = "application/json"
+)
+
+// IsCloudEventContentType checks for content type.
+func IsCloudEventContentType(contentType string) bool {
+	return isContentType(contentType, CloudEventContentType)
+}
+
+// IsJSONContentType checks for content type.
+func IsJSONContentType(contentType string) bool {
+	return isContentType(contentType, JSONContentType)
+}
+
+func isContentType(contentType string, expected string) bool {
+	lowerContentType := strings.ToLower(contentType)
+	if lowerContentType == expected {
+		return true
+	}
+
+	semiColonPos := strings.Index(lowerContentType, ";")
+	if semiColonPos >= 0 {
```

Comment:
Is there a reason why this can't be changed to:

```go
return strings.Contains(lowerContentType, expected)
```

Reply:
To avoid false positives where one content type is a substring of another but not the same.

Example: `application/epub` is a substring of `application/epub+zip` but not the same.

Reply:
I would argue that in this case they are the same as far as content type matching is concerned, but I don't mind to keep this as is.